### DISCUSSION TITLE
Directionality toggle for textual fields

### DISCUSSION THREAD

**Discussion by proimage** [craftcms/cms#9639] [Ideas]
Ref #9616

It would make sense to be able to override the directionality/language of specific fields to something other than the site language. For example, even in an RTL site, with an RTL user preference, text fields for computer code (for display or for embedding) should 99.99% of the time be LTR.

I can't think of a specific use case for a directionality toggle on fields like dropdowns and the like, only for textual input fields, but I'm not sure if implementing it for textual fields only is a good thing or not.

Now that I think about it though, it might make sense to be able to specify the _language_ of a field (which would supersede the directionality toggle). For example, a scholarly site showing translations of a sentence in multiple languages on the same page. That's not something that multi-site would be able to help with, is it?

**Comment by proimage** [comment]
@brandonkelly So this is biting me in the rear on a currently in-development project. The site's primary content is RTL, but there's all sorts of fields for things like telephone numbers and URLs that get really, really messed up when rendered as RTL.

Worth noting is that said fields are not _just_ the normal Craft fieldtypes, but also things like the 3rd-party Typed Link and Phone Number fields. I'm not sure which is the "proper" way to resolve this, but from my POV it would be great if this were something that was defined and addressed at the CMS level, not at the level of individual plugins.

In other words, on an RTL site, if a custom field could be set to LTR (through a control that the CMS provides), then ideally that custom field wouldn't really be aware that the site is RTL at all.

**Reply by brandonkelly** [reply]
In the meantime it might be something you could fix with the [Control Panel CSS](https://plugins.craftcms.com/cp-css) plugin.

**Reply by proimage** [reply]
For future reference of anyone stumbling across this and looking for a CP CSS solution, here's what I have for a text field whose handle is `blockId`:

```css
body.ltr [id$="-blockId-field"] .rtl .text
{
	text-align: left !important;
	direction: ltr !important;
	padding-left: 9px !important;
	padding-right: 28.4px !important;
}
body.ltr [id$="-blockId-field"] .input.rtl > .chars-left {
	float: right !important;
	margin-left: 0 !important;
}
```

Not ideal, with lots of "magic numbers" used to "un-mirror" Craft's RTL tweaks, but it works for me. I think the thing that bothers me the most though is the use of `!important` in Craft's own control panel CSS: why, oh why? 😭 The same exact RTL styles work just fine without using `!important`...
